src/Periphery/Permit2Proxy.sol

```diff
@@ -0,0 +1,356 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.17;
+
+import { ISignatureTransfer } from "permit2/interfaces/ISignatureTransfer.sol";
+import { TransferrableOwnership } from "lifi/Helpers/TransferrableOwnership.sol";
+import { LibAsset, IERC20 } from "lifi/Libraries/LibAsset.sol";
+import { PermitHash } from "permit2/libraries/PermitHash.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { ERC20Permit } from "@openzeppelin/contracts/token/ERC20/extensions/ERC20Permit.sol";
+
+/// @title Permit2Proxy
+/// @author LI.FI (https://li.fi)
+/// @notice Proxy contract allowing gasless (Permit2-enabled) calls to our
+///         diamond contract
+/// @custom:version 1.0.0
+contract Permit2Proxy {
+    /// Storage ///
+
+    address public immutable LIFI_DIAMOND;
+    ISignatureTransfer public immutable PERMIT2;
+    mapping(address => bool) public diamondWhitelist;
```

Comment:
it seems that the whitelist handling (adding/removing) has been removed from this contract. If we are using the whitelist, then we need to add functions to manipulate it and some sort of admin-only access control.